src/HttpUserAgentParser/HttpUserAgentParser.cs

```diff
@@ -198,39 +208,43 @@ private static bool TryExtractVersion(ReadOnlySpan<char> haystack, out Range ran
 
         // Limit search window to avoid scanning entire UA string unnecessarily
         const int Window = 128;
-        if (haystack.Length >= Window)
+        if (haystack.Length > Window)
         {
             haystack = haystack.Slice(0, Window);
         }
 
-        int i = 0;
-        for (; i < haystack.Length; ++i)
+        // Find first digit
+        int start = -1;
+        for (int i = 0; i < haystack.Length; i++)
         {
             char c = haystack[i];
-            if (char.IsBetween(c, '0', '9'))
+            if (c >= '0' && c <= '9')
```

Comment:
This is two comparisons now, instead of just one. Either use the `uint` cast directly, or use `char.IsBetween` which does the same thing but more readable IMO.

Only the JIT for .NET 10+ is able to optimize that to one comparison (by using the `uint` cast).